statsmodels/multivariate/factor.py

```diff
@@ -230,34 +279,261 @@ def _fit_pa(self, maxiter=50, tol=1e-8):
+
+        # log|GG' + S|
+        # Using matrix determinant lemma:
+        # |GG' + S| = |I + G'S^{-1}G|*|S|
+        lul.flat[::lul.shape[0]+1] += 1
+        _, ld = np.linalg.slogdet(lul)
+        v = np.sum(np.log(uniq)) + ld
+
+        # tr((GG' + S)^{-1}C)
+        # Using Sherman-Morrison-Woodbury
+        w = np.sum(1 / uniq)
+        b = np.dot(load.T, self.corr / uniq[:, None])
+        b = np.linalg.solve(lul, b)
+        b = np.dot(loadu, b)
+        w -= np.trace(b)
+
+        # Scaled log-likelihood
+        return -(v + w) / (2*self.k_endog)
+
+    def score(self, par):
+        """
+        Evaluate the score function.
+
+        Arguments
+        ---------
+        par : ndarray or tuple of 2 ndarray's
+            The model parameters, either a packed representation of
+            the model parameters or a 2-tuple containing a `k_endog x
+            n_factor` matrix of factor loadings and a `k_endog` vector
+            of uniquenesses.
+        """
+
+        if type(par) is np.ndarray:
+            uniq, load = self._unpack(par)
+        else:
+            load, uniq = par[0], par[1]
+
+        # Center term of SMW
+        loadu = load / uniq[:, None]
+        c = np.dot(load.T, loadu)
+        c.flat[::c.shape[0]+1] += 1
+        d = np.linalg.solve(c, load.T)
+
+        # Precompute these terms
+        lud = np.dot(loadu, d)
+        cu = (self.corr / uniq) / uniq[:, None]
+        r = np.dot(cu, load)
+        lul = np.dot(lud.T, load)
+        luz = np.dot(cu, lul)
+
+        # First term
+        du = 2*np.sqrt(uniq) * (1/uniq - (d * load.T).sum(0) / uniq**2)
+        dl = 2*(loadu - np.dot(lud, loadu))
+
+        # Second term
+        h = np.dot(lud, cu)
+        f = np.dot(h, lud.T)
+        du -= 2*np.sqrt(uniq) * (np.diag(cu) - 2*np.diag(h) + np.diag(f))
+        dl -= 2*r
+        dl += 2*np.dot(lud, r)
+        dl += 2*luz
+        dl -= 2*np.dot(lud, luz)
+
+        # Can't use _pack because we are working with the square root
+        # uniquenesses directly.
+        return -np.concatenate((du, dl.T.flat)) / (2*self.k_endog)
+
+    # Maximum likelihood factor analysis.
+    def _fit_ml(self, start, em_iter, opt_method, opt):
+
+        # Starting values
+        if start is None:
+            load, uniq = self._fit_ml_em(em_iter)
+            start = self._pack(load, uniq)
+        elif len(start) == 2:
+            if len(start[1]) != start[0].shape[0]:
+                msg = "Starting values have incompatible dimensions"
+                raise ValueError(msg)
+            start = self._pack(start[0], start[1])
+        else:
+            raise ValueError("Invalid starting values")
+
+        def nloglike(par):
+            return -self.loglike(par)
+
+        def nscore(par):
+            return -self.score(par)
+
+        # Do the optimization
+        if opt is None:
+            opt = _opt_defaults
+        r = minimize(nloglike, start, jac=nscore, method=opt_method,
```

Comment:
you could add `mle_retvals` to the results class, so it's possible to inspect convergence information.